src/client_side.cc

```diff
@@ -1655,8 +1653,6 @@ clientProcessRequest(ConnStateData *conn, const Http1::RequestParserPointer &hp,
             request->flags.disableCacheUse("cache manager URL");
     }
 
-    request->flags.internal = http->flags.internal;
```

Comment:
Disclaimer: I have not checked all the low-level details, especially whether the removal of this assignment may change Squid behavior in some (hypothetical) cases where `http->flags.internal` was false but `request->flags.internal` was true (and, hence, became false here in the official code, due to this assignment).